Sources/Lifecycle/Lifecycle.swift

```diff
@@ -590,25 +592,27 @@ public class ComponentLifecycle: LifecycleTask {
         }
     }
 
-    private func startTask(on queue: DispatchQueue, tasks: [LifecycleTask], index: Int, callback: @escaping (Int, Error?) -> Void) {
+    private func startTask(on queue: DispatchQueue, tasks: [LifecycleTask], index: Int, callback: @escaping ([LifecycleTask], Error?) -> Void) {
         // async barrier
         let start = { (callback) -> Void in queue.async { tasks[index].start(callback) } }
         let callback = { (index, error) -> Void in queue.async { callback(index, error) } }
 
         if index >= tasks.count {
-            return callback(index, nil)
+            return callback(tasks, nil)
         }
         self.logger.info("starting tasks [\(tasks[index].label)]")
         let startTime = DispatchTime.now()
         start { error in
             Timer(label: "\(self.label).\(tasks[index].label).lifecycle.start").recordNanoseconds(DispatchTime.now().uptimeNanoseconds - startTime.uptimeNanoseconds)
             if let error = error {
                 self.logger.error("failed to start [\(tasks[index].label)]: \(error)")
-                return callback(index, error)
+                let started = Array(tasks.prefix(index))
+                return callback(started, error)
             }
             // shutdown called while starting
             if case .shuttingDown = self.stateLock.withLock({ self.state }) {
-                return callback(index, nil)
+                let started = index < tasks.count ? Array(tasks.prefix(index + 1)) : tasks
```

Comment:
Is the task at `index` considered started?

Reply:
this is a non-error case, meaning the task was started successfully. do you see something else?